### ISSUE TITLE
No error on invalid descriptor `__get__` signature

### ISSUE THREAD

**Issue by sharkdp** [astral-sh/ty#4129]
ty should emit an error here on the attribute access, or on the invalid `__get__` signature:

```py
class Descriptor:
    def __get__(self): pass

class C:
    attr = Descriptor()

C().attr  # Should be an error
```

https://play.ty.dev/1e6b5165-78b0-40f7-b49c-deecc0a840db

This is an error at runtime and all other type checkers flag this.
